### PR TITLE
reduce CI compile time for atomicPhysics compile test

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- avoid compiling derived fields (added fileOutput.param)
- remove unused particle and density functors.

# Motivation

There is no need that our compile test contains functors which will be not used in the example or compiles more than required to test atomic physics compile.

### CODE REVIEW

**Review by BrianMarre — APPROVED** [review]
Looks good!

File: share/picongpu/tests/compileAtomicPhysics/include/picongpu/param/particle.param [hunk 1]
```diff
@@ -46,19 +46,6 @@ namespace picongpu::particles
...
-            /** Count of particles per cell at initial state
-             *  unit: none
-             */
-            static constexpr uint32_t numParticlesPerCell = 1u;
-            // sit directly in lower corner of the cell
-            // each x, y, z in-cell position component in range [0.0, 1.0)
-            static constexpr auto inCellOffset = float3_X(0.0, 0.0, 0.0);
-        };
-        using OnePosition = OnePositionImpl<OnePositionParameter>;
-        using Random10ppc = RandomImpl<RandomParameter10ppc>;
```
Comment (BrianMarre) [inline share/picongpu/tests/compileAtomicPhysics/include/picongpu/param/particle.param:61]: Line 61 is still required

**Review by BrianMarre — CHANGES_REQUESTED** [review]
@psychocoderHPC you also removed the definition of the 10ppc random particle functor used in the init pipeline

File: share/picongpu/tests/compileAtomicPhysics/include/picongpu/param/particle.param (hunk 1 above)
Comment (PrometheusPi) [reply share/picongpu/tests/compileAtomicPhysics/include/picongpu/param/particle.param:61]: @BrianMarre is right about that. The `Random10ppc` is used [here](https://github.com/ComputationalRadiationPhysics/picongpu/blob/2572a435a2b0a1e2121ee12cd885a9817ea0c571/share/picongpu/tests/compileAtomicPhysics/include/picongpu/param/speciesInitialization.param#L44). It is surprising for me that the CI still passes as this should cause a compile error. Do we execute this test? (EDIT: CI fails, didn't see that before)

**Review by PrometheusPi — CHANGES_REQUESTED** [review]
One bug

File: share/picongpu/tests/compileAtomicPhysics/include/picongpu/param/particle.param (hunk 1 above)
Comment (psychocoderHPC) [reply share/picongpu/tests/compileAtomicPhysics/include/picongpu/param/particle.param:61]: re-added it

**Review by BrianMarre — APPROVED** [review]

**Review by PrometheusPi — APPROVED** [review]

**Pull request merged**